events/class_tests/create_meeting_form.py

```diff
@@ -0,0 +1,195 @@
+import pytest
+from users.models import User
+from events.models import Event, Colors, EventParticipant, OptionalMeetingDates
+from events.views import CreateMeetingView
+from events.forms import EventCreationForm
+from datetime import datetime
+from django.utils import timezone
+from pytest_django.asserts import assertTemplateUsed
+
+TITLE = 'new_form'
+LOCATION = 'new_form_location'
+DESCRIPTION = 'new_form_description'
+EVENT_DATE_TIME_START = datetime(2023, 11, 11, 12, 12, 12, 0, tzinfo=timezone.utc)
+EVENT_DATE_TIME_END = datetime(2023, 11, 11, 14, 12, 12, 0, tzinfo=timezone.utc)
+COLOR = Colors.BLACK
+
+OPTIONAL_MEETING_DATE_START = datetime(2023, 3, 24, 12, 12, 12, 0, tzinfo=timezone.utc)
+OPTIONAL_MEETING_DATE_END = datetime(2023, 3, 24, 14, 12, 12, 0, tzinfo=timezone.utc)
+OPTIONAL_MEETING_DATE_START_IN_THE_PAST = "2020-11-11 10:10"
+OPTIONAL_MEETING_DATE_END_IN_THE_PAST = "2020-11-11 11:40"
+
+MEETING_CREATION_URL = '/event/meeting/'
+MEETING_CREATION_HTML_PATH = "meetings/create_meeting.html"
+
+
+@pytest.fixture
+def valid_event_data():
+    return {
+        "title": TITLE,
+        "location": LOCATION,
+        "description": DESCRIPTION,
+        "date_time_start": EVENT_DATE_TIME_START,
+        "date_time_end": EVENT_DATE_TIME_END,
+        "color": COLOR,
+    }
+
+
+@pytest.fixture
+def valid_optional_meeting_data():
+    return {
+        'optional_meetings-TOTAL_FORMS': '1',
+        'optional_meetings-INITIAL_FORMS': '0',
+        'optional_meetings-0-date_time_start': OPTIONAL_MEETING_DATE_START,
+        'optional_meetings-0-date_time_end': OPTIONAL_MEETING_DATE_END,
+    }
+
+
+@pytest.fixture
+def valid_participant_data():
+    return {
+        'participants-TOTAL_FORMS': '1',
+        'participants-INITIAL_FORMS': '0',
+        'participants-0-participant_email': 'testUser2@mta.ac.il',
+    }
+
+
+@pytest.mark.django_db
+class TestCreateMeetingForm:
+    @pytest.fixture
+    def signed_up_user_details(self):
+        return {'email': 'testUser1@mta.ac.il', 'password': 'PasswordU$er123'}
+
+    @pytest.fixture
+    def sign_in(self, client, signed_up_user_details):
+        return client.post('/login/', data=signed_up_user_details)
+
+    def test_meeting_created(self, sign_in, valid_event_data, valid_optional_meeting_data, valid_participant_data,
+                             client):
+        user_id = User.objects.get(pk=1)
+        request = client.get(MEETING_CREATION_URL)
+        view = self.create_meeting_view(user_id, valid_event_data, valid_optional_meeting_data, valid_participant_data)
```

Comment:
Can you explain why aren't you using client.post instead of creating a meeting view and self assigning variables of the class?

Reply:
IMHO the 'post' method of 'CreateMeetingView' includes alot of logic, therefore i created it manually in order to be able to know exactly where the test fails in case it happen.

Reply:
You can use client.post
and in the response context, you can access the form errors as it printed to the user

the format is like this: 
response = client.post(url, data)
response.context['form'].errors ...

I think this is a simpler way to test the needed tests because you only check the error text
but this is your choice if you want to change 😄 
each decision is right by me